### DISCUSSION TITLE
When the user drags an element out of bounds how do I translate the paper in that direction

### DISCUSSION THREAD

**Discussion by IshaanShettigar** [clientIO/joint#2363] [Q&A] [answered]
### Introduction

Imagine we have an infinite paper. The user drags an element to the top of the screen, I would like to automatically translate the paper in the direction that the element is being dragged. I thought of listening for the `element.on(change:position)` or `graph.on(change:position)` but they only give me access to the new position and the element. There is no way for me to get the actual mouse coordinates and detect whether it is at the edge of the screen and then trigger the `paper.translate()`.

Any suggestions on how I should proceed?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
LIsten to the paper's `element:pointermove` event instead. There you receive the mouse `event` as the second parameter.

**Note:** The `ui.PaperScroller` plugin of *JointJS+* already implements this feature. It can enabled with `scrollWhileDragging`.
